File: leetcode/1512/solution.js
/**
 * https://leetcode.com/problems/number-of-good-pairs/
 * 1512. Number of Good Pairs
 */

/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function (nums) {
  const arr = new Array(101).fill(0);
  let count = 0;

  for (const n of nums) {
    count += arr[n];
    arr[n] += 1;
  }
  return count;
};

File: leetcode/1512/solution.rs
/**
 * https://leetcode.com/problems/number-of-good-pairs/
 * 1512. Number of Good Pairs
 */
impl Solution {
    pub fn num_identical_pairs(nums: Vec<i32>) -> i32 {
        let mut counter: [i32; 101] = [0; 101];
        let mut count = 0;
        for n in nums {
            let n = n as usize;
            count += counter[n];
            counter[n] += 1;
        }
        count
    }
}
